### ISSUE TITLE
Implement HasAriaLabel interface in Checkbox

### ISSUE THREAD

**Issue by web-padawan** [vaadin/flow-components#4718]
### Describe your motivation

Part of #2946
Part of vaadin/platform#3803

### Describe the solution you'd like

1. Implement `HasAriaLabel` interface in `Checkbox` component when the label is missing,
2. Add unit tests ensuring that the attribute value overrides the one from the web compoent.
